File: src/main/python/g0001_0100/s0009_palindrome_number/Solution.py
# #Easy #Math #Udemy_Integers #Top_Interview_150_Math #Big_O_Time_O(log10(x))_Space_O(1)
# #2025_07_22_Time_4_ms_(80.28%)_Space_17.98_MB_(19.56%)

class Solution:
    def isPalindrome(self, x: int) -> bool:
        return str(x) == str(x)[::-1]

File: src/main/js/g0001_0100/s0009_palindrome_number/solution.js
// #Easy #Math #Udemy_Integers #Top_Interview_150_Math #Big_O_Time_O(log10(x))_Space_O(1)
// #2024_12_03_Time_4_ms_(97.14%)_Space_59.8_MB_(5.70%)

/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
    if (x < 0) {
        return false
    }

    let rev = 0
    let localX = x

    while (localX > 0) {
        rev = rev * 10 + (localX % 10)
        localX = Math.floor(localX / 10)
    }

    return rev === x
}

export { isPalindrome }
